packages/lodestar/src/network/reqresp/reqResp.ts

```diff
@@ -215,26 +222,45 @@ export class ReqResp implements IReqResp {
 
     switch (requestTyped.method) {
       case Method.Ping:
+        if (!this.inboundRateLimiter.allowRequest(peerId)) {
+          throw new ResponseError(RespStatus.RATE_LIMITED, "rate limit");
+        }
         yield this.metadataController.seqNumber;
         break;
       case Method.Metadata:
+        if (!this.inboundRateLimiter.allowRequest(peerId)) {
+          throw new ResponseError(RespStatus.RATE_LIMITED, "rate limit");
+        }
         // V1 -> phase0, V2 -> altair. But the type serialization of phase0.Metadata will just ignore the extra .syncnets property
         // It's safe to return altair.Metadata here for all versions
         yield this.metadataController.json;
         break;
       case Method.Goodbye:
+        // no need to call this.inboundRateLimiter.allowRequest
+        this.inboundRateLimiter.prune(peerId);
```

Comment:
Rely exclusively on the libp2p disconnect event for pruning. A goodbye is not guaranteed to happen